src/ca/team2706/vision/trackerboxreloaded/Main.java

```diff
@@ -41,13 +46,60 @@
 
 	/**
 	 * A class to hold any vision data returned by process()
+	 * :)
+	 * :)
+	 * :}
+	 * :]
+	 * :]
 	 */
 	public static class VisionData {
 		public Mat outputImg = new Mat();
-		double fps;
+		public double fps;
+		public HashMap<String,String> data = new HashMap<String,String>();
+		/**
+		 * This method converts the vision data into a nice and tidy string
+		 * :]
+		 * @return the data
+		 */
+		public void encode(NetworkTable table){
+			//TODO: add code
+			//basic networktable code: table.getEntry("data").setString("foo);
+			table.getEntry("data").setString(DataUtils.encodeData(data));
+		}
+		
+		/**
+		 * Decodes the NetworkTable stuff into a VisionData
+		 * @param the data table
+		 * @return the data
+		 */
+		@SuppressWarnings("unused")
+		public static VisionData decode(NetworkTable table){
+			//TODO: add code
+			//basic networktable code: table.getEntry("data").getValue().getString();
+			//dont forget: if(table.getEntry("data").getValue().isString())
+			if(table.getEntry("data").getValue().isString()){
+				HashMap<String,String> data = DataUtils.decodeData(table.getEntry("data").getValue().getString());
+			}
+			return null;
+		}
```

Comment:
Why would the vision code on the pi ever need to read vision data from the NetworkTable?